irods/manager/user_manager.py

```diff
@@ -82,6 +83,72 @@ def temp_password_for_user(self, user_name):
             return obf.create_temp_password(msg.stringToHashWith, conn.account.password)
 
 
+    class EnvStoredPasswordNotEdited(RuntimeError):
+
+        """
+        Error thrown by a password change attempt if a login password encoded in the
+        irods environment could not be updated.
+
+        This error will be seen when `modify_secrets_file' is set True and the
+        authentication scheme in effect for the session is other than iRODS native,
+        using a password loaded from the client environment.
+	"""
+
+        def __init__(self,msg=None):
+            """Initialize  the exception object (with a message if desired)."""
+            super(UserManager.EnvStoredPasswordNotEdited,self).__init__(msg)
+
+    @staticmethod
+    def abspath_exists(path):
+        return (isinstance(path,str) and
+                os.path.isabs(path) and
+                os.path.exists(path))
+
+    def modify_own_password(self, old_value, new_value, modify_secrets_file = False):
+
+        '''
+        Change the password for the current user (in the manner of `ipasswd').
+
+        Parameters:
+            old_value - the currently valid (old) password
+            new_value - the desired (new) password
+            modify_secrets_file - Can be False, True, or a string.  If a string, it should indicate
```

Comment:
consider `modify_irods_authentication_file` instead.    we don't call it a secrets file anywhere else.

Reply:
done